pkg/miniogw/test/integration_test.go

```diff
@@ -0,0 +1,204 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package integration_test
+
+import (
+	"context"
+	"flag"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/minio/cli"
+	minio "github.com/minio/minio/cmd"
+	"github.com/stretchr/testify/assert"
+	"github.com/zeebo/errs"
+
+	monkit "gopkg.in/spacemonkeygo/monkit.v2"
+
+	"storj.io/storj/internal/s3client"
+	"storj.io/storj/internal/testplanet"
+	"storj.io/storj/pkg/cfgstruct"
+	"storj.io/storj/pkg/miniogw"
+	"storj.io/storj/pkg/miniogw/logging"
+	"storj.io/storj/pkg/provider"
+)
+
+const (
+	apiKey = "apiKey"
+	encKey = "encKey"
+)
+
+var (
+	mon = monkit.Package()
+	//Error is the errs class of standard End User Client errors
+	Error = errs.Class("Storj Gateway error")
+
+	ctx = context.Background()
+)
+
+func TestUploadDownload(t *testing.T) {
+	tests := []struct {
```

Comment:
There's a single long case, no need for the table, this makes it harder to read.